internal/js/bundle.go

```diff
@@ -164,13 +164,16 @@ func NewBundleFromArchive(piState *lib.TestPreInitState, arc *lib.Archive) (*Bun
 }
 
 func (b *Bundle) makeArchive() *lib.Archive {
+	clonedSourceDataURL, _ := url.Parse(b.sourceData.URL.String())
+	clonedPwdURL, _ := url.Parse(b.pwd.String())
+
 	arc := &lib.Archive{
 		Type:              "js",
 		Filesystems:       b.filesystems,
```

Comment:
Other fields here, like this `Filesystems` are also passed by reference (note it's a `map`), and thus are likely to be modified. However, that doesn't seem to cause any issue so far.